### PR TITLE
test: Fix intermittent failures with ReqwestClient

### LINKED ISSUE #259: Intermittent test failure with ReqwestClient
This failure was intermittently observed both in CI and locally with the different test cases.

Case I:
```
ERRORmerino_adm::remote_settings::reqwest_client: ReqwestClient - unable to submit GET request. "Performing HTTP request for Remote Settings: http://localhost:8888/buckets/monitor/collections/changes/changeset?_expected=0", type: "adm.remote-settings.reqwest.get-failed"
    at merino-adm/src/remote_settings/reqwest_client.rs:45

thread 'suggest::suggest_adm_rs_works_empty' panicked at 'Could not create providers: There was a network error while setting up this suggestions provider
``` 

Case II:
```
ERRORmerino_adm::remote_settings::reqwest_client: ReqwestClient - unable to submit GET request. "Performing HTTP request for Remote Settings: http://localhost:8888/", type: "adm.remote-settings.reqwest.get-failed"
    at merino-adm/src/remote_settings/reqwest_client.rs:45
    in merino_adm::remote_settings::sync with suggestion_score: Proportion(1216271616)

thread 'suggest::suggest_adm_rs_score_is_configurable' panicked at 'Could not create providers: There was a network error while setting up this suggestions provider
```

### PR DESCRIPTION
**Pull request by ncloudioj**
This fixes #259.

The issue was caused by the connection pool in Reqwest/Hyper. See more details in [here](https://github.com/hyperium/hyper/issues/2136#issuecomment-589345238).

Since `ReqwestClient` is only used by RS suggester, and it is not making constant network calls to RS. Disabling the connection pool should not be a big deal, but I will let @mythmon @Dexterp37 decide if this is acceptable or shall we keep it as is and add the retry to the API.

### CODE REVIEW

File: merino-adm/src/remote_settings/reqwest_client.rs [hunk 1]
```diff
@@ -20,10 +20,13 @@ pub struct ReqwestClient {
 
 impl ReqwestClient {
     /// Instantiate a new Reqwest client to perform HTTP requests.
-    pub fn new() -> ReqwestClient {
-        Self {
-            reqwest_client: reqwest::Client::new(),
```
Comment (ncloudioj) [inline merino-adm/src/remote_settings/reqwest_client.rs:25]: Just noticed that `request::Client::new()` could panic in certain cases, but the builders let you capture those errors without panicking.

File: merino-adm/src/remote_settings/reqwest_client.rs [hunk 2]
```diff
@@ -20,10 +20,13 @@ pub struct ReqwestClient {
...
-        }
+    pub fn try_new() -> Result<ReqwestClient, Error> {
+        let reqwest_client = reqwest::Client::builder()
+            // Disable the connection pool to avoid the IncompleteMessage errors.
+            // See #259 for more details.
+            .pool_max_idle_per_host(0)
```
Comment (Dexterp37) [inline merino-adm/src/remote_settings/reqwest_client.rs:27]: I'm wondering why this wasn't happening with the other implementation: we were not setting the max idle per host there as well. Any idea?

Aside from that, I don't think there's a big problem in disabling pooling.

**Review by mythmon — APPROVED** [review]
I think a better thing to do would be to change the time that `reqwest` keeps connections around so it doesn't outlive the Kinto connections, but I can't find what that value would be. After that, I think adding retry is going to be an important long term solution for many parts of Merino. Neither of these are quick though.

This unblocks us now, and is simple enough to understand. Lets file an issue to come back to this, and merge this fix.

**Comment by pjenvey** [comment]
I'll note that [hyper v0.14.3](https://github.com/hyperium/hyper/blob/master/CHANGELOG.md#v01413-2021-09-16) has a fix around connection reuse + flushing but it definitely does not help, I tried it as a long shot [here](https://app.circleci.com/pipelines/github/mozilla-services/merino/706/workflows/7d89d598-3611-4a1d-979b-bb0c2b660cef/jobs/3209)

**Review by pjenvey — APPROVED** [review]

**Pull request merged**